src/payment/data/sagas.test.js

```diff
@@ -863,6 +872,225 @@ describe('saga tests', () => {
+      }));
+
+    const successfulTestPlan = baseStateTable.map((lineItem) => ({
+      stateKey: lineItem.stateKey,
+      stateValue: lineItem.stateValue,
+      validToPoll: lineItem.validToPoll,
+      inputState: generateTestBasketState(lineItem.stateValue, [{ paymentNumber: 'ABCXYZ1' }]),
+      shouldFail: false,
+      expectedResult: lineItem.validToPoll ? expectPollAndSucceed() : expectQuickSuccess(),
+    }));
+
+    const runtimeFailureTestPlan = baseStateTable.filter((lineItem) => lineItem.validToPoll)
+      .map((lineItem) => ({
+        stateKey: lineItem.stateKey,
+        stateValue: lineItem.stateValue,
+        validToPoll: lineItem.validToPoll,
+        inputState: generateTestBasketState(lineItem.stateValue, [{ paymentNumber: null }]),
+        shouldFail: true,
+        expectedResult: expectRuntimeFailure(),
+      }));
+
+    const testPlan = [...successfulTestPlan, ...runtimeFailureTestPlan];
+
+    for (let i = 0, test = testPlan[i]; testPlan.length > i; i++, test = testPlan[i]) {
+      const nt = test.validToPoll ? '' : 'n\'t';
+      const suffix = test.shouldFail ? ', but should fail becase of a missing value' : ', and succeed';
+      it(`PAYMENT_STATE.${test.stateKey} should${nt} poll${suffix}`, async () => {
+        const localDispatched = [];
+        const localCaughtErrors = [];
+        axiosMock.reset();
+
+        axiosMock.onGet(CC_PAYMENT_STATE_ENDPOINT).reply(
+          test.shouldFail ? 404 : 200,
+          { state: PAYMENT_STATE.COMPLETED },
+        );
+
+        const localSagaOptions = {
+          dispatch: action => localDispatched.push(action),
+          onError: err => localCaughtErrors.push(err),
+        };
+
+        try {
+          await runSaga(
+            {
+              getState: () => ({
+                payment:
+                    {
+                      basket: test.inputState,
+                    },
+              }),
+              ...localSagaOptions,
+            },
+            handlePaymentState,
+          ).toPromise();
+        } catch (e) {} // eslint-disable-line no-empty
+
+        expect(localDispatched).toEqual(test.expectedResult);
+        expect(localCaughtErrors).toEqual([]);
+      });
+    }
+
+    it(`Should Retry HTTP Errors ${DEFAULT_PAYMENT_STATE_POLLING_MAX_ERRORS}x then fail`, async () => {
+      mergeConfig({
+        PAYMENT_STATE_POLLING_DELAY_SECS: 0.1,
+      });
+
+      const localDispatched = [];
+      const localCaughtErrors = [];
+      axiosMock.reset();
+
+      axiosMock.onGet(CC_PAYMENT_STATE_ENDPOINT).reply(404, null);
+
+      const editableState = ({
+        payment:
+            {
+              basket: generateTestBasketState(PAYMENT_STATE.PENDING, [{ paymentNumber: 'QAZWSX' }]),
+            },
+      });
+
+      const localSagaOptions = {
+        dispatch: action => {
+          localDispatched.push(action);
+          editableState.payment.basket = paymentState(editableState.payment.basket, action);
```

Comment:
This is how you invoke the reducer to properly update state when executing a saga that requires the proper reducer mechanics. Without this call and the `editableState` above, things like our error countdown fails.